### PR TITLE
HELM-358: Document panels

### PR DESCRIPTION
**Pull request by emarsh1995**
- Update `panel_configuration/index.adoc`
- Update `ROOT/nav.adoc`
- Update `local-site.yml`, `site.yml`
- Add `panel_configuration/alarm-histogram-example.png`, `/alarms-table-example.png`, `/flow-histogram-example.png`

# Pull Request Check Sheet

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] ~~If this is a new feature, is there documentation?~~
* [ ] ~~If this is a new feature or substantial change, are there tests that cover it?~~

# Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

Once the reviewer(s) accept the PR and the branch passes continuous integration in Bamboo, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR. Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

# External References

* JIRA (Issue Tracker): https://opennms.atlassian.net/browse/HELM-358
* Continuous Integration: [CircleCI](https://circleci.com/gh/OpenNMS/opennms-helm)

### CODE REVIEW

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [inline docs/modules/panel_configuration/pages/index.adoc:28]: I would mention that the alarms table also can display any information associated to the alarm, like status, location, node assciated info, assets associated info (whatever is allowed in the panel options) that the user may find relevant. And also their status can also be interacted from the panel (like aknowledge the alarm, clear the alarm, add memos, etc)

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [inline docs/modules/panel_configuration/pages/index.adoc:33]: I would mention that the alarms table also can display any information associated to the alarm, like status, location, node assciated info, assets associated info (whatever is allowed in the panel options) that the user may find relevant. And also their status can also be interacted from the panel (like aknowledge the alarm, clear the alarm, add memos, etc)

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [inline docs/modules/panel_configuration/pages/index.adoc:29]:
```suggestion

```
This is only for Flows Histogram

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [inline docs/modules/panel_configuration/pages/index.adoc:42]: This are the minimum options required to deisplay some data. Users can add any other function/transformation

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [inline docs/modules/panel_configuration/pages/index.adoc:82]:
```suggestion
. Choose a datasource from the *Data Source* list (in this example, *OpenNMS Entities*).
```
Ah the **React** text is during this migration we are working on, but it shouldn't be part of the documentation since it will be removed once we merged into develop.

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [inline docs/modules/panel_configuration/pages/index.adoc:94]:
```suggestion
. Add, remove, and reorder columns in the Alarm Table section.
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [inline docs/modules/panel_configuration/pages/index.adoc:107]:
```suggestion
. Choose a datasource from the *Data Source* list (in this example, *OpenNMS Flow*).
```

**Review by aramosvizcarra — DISMISSED** [review]
A few minor changes and suggestion but is looking pretty good. 
I'm sure @mmahacek can give you some good feedback since he knows Grafana and Helm backwards and forward!

File: local-site.yml
Comment (mmahacek) [inline local-site.yml:12]:
```suggestion
    url: https://github.com/opennms-forge/antora-ui-opennms/releases/download/v3.1.0/ui-bundle.zip
```

File: site.yml
Comment (mmahacek) [inline site.yml:13]:
```suggestion
    url: https://github.com/opennms-forge/antora-ui-opennms/releases/download/v3.1.0/ui-bundle.zip
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (mmahacek) [inline docs/modules/panel_configuration/pages/index.adoc:33]: How does one configure a Filter Panel? What other kinds of panels will it apply to? How do you configure the other panels to honor the filter panel's selection?

File: docs/modules/panel_configuration/pages/index.adoc
Comment (mmahacek) [inline docs/modules/panel_configuration/pages/index.adoc:20]:
```suggestion
== Included custom panels
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (aramosvizcarra) [reply docs/modules/panel_configuration/pages/index.adoc:33]: adding @synqotik here since he worked on the filter panel and might have a better answer than me. Although my understanding is Filter Panel only works with Alarm Table Panel where an alarm table check if there is a Filter panel in the dashboard and apply the filter to the Alarm Table panel....but I'm not too sure about that

**Review by Bonrob2 — COMMENTED** [review]
Minor feedback. I'd like to review again when everything is updated so I can build it locally and review. Thank you!

File: docs/modules/panel_configuration/pages/index.adoc
Comment (synqotik) [reply docs/modules/panel_configuration/pages/index.adoc:33]: I'll take a look at this, with respect to the Filter Panel.

File: docs/modules/panel_configuration/pages/index.adoc
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:46]:
```suggestion
For a panel to display data visualizations, you must define the following settings:

* Datasource
* Flow Type
* Time Interval

You can also add any other function or transformation to customize the visualized data.
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:87]:
```suggestion
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:85]:
```suggestion
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (emarsh1995) [inline docs/modules/panel_configuration/pages/index.adoc:87]:
```suggestion
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (Bonrob2) [inline docs/modules/panel_configuration/pages/index.adoc:51]:
```suggestion
You can customize the data visualized on {page-component-title} panels by editing their settings and changing their datasources.
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (Bonrob2) [inline docs/modules/panel_configuration/pages/index.adoc:56]:
```suggestion
Follow these steps to customize an alarm histogram panel:
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (Bonrob2) [inline docs/modules/panel_configuration/pages/index.adoc:58]:
```suggestion
. Enter editing mode and select an alarm histogram panel on your dashboard.
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (Bonrob2) [inline docs/modules/panel_configuration/pages/index.adoc:80]:
```suggestion
Follow these steps to customize an alarms table panel:
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (Bonrob2) [inline docs/modules/panel_configuration/pages/index.adoc:82]:
```suggestion
. Enter editing mode and select an alarms table panel on your dashboard.
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (Bonrob2) [inline docs/modules/panel_configuration/pages/index.adoc:102]:
```suggestion
Follow these steps to customize a flow histogram panel:
```

File: docs/modules/panel_configuration/pages/index.adoc
Comment (Bonrob2) [inline docs/modules/panel_configuration/pages/index.adoc:104]:
```suggestion
. Enter editing mode and select a flow histogram panel on your dashboard.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few minor changes. Thanks!

File: docs/modules/panel_configuration/pages/index.adoc
Comment (synqotik) [reply docs/modules/panel_configuration/pages/index.adoc:32]: Filter Panel overview above should read something like "A panel that allows you to filter entity data in other panels"

Comment (synqotik) [reply docs/modules/panel_configuration/pages/index.adoc:32]: I could help write up some documentation for the Filter Panel. Not sure if it's better to do it here, or else just open a separate ticket for it rather than hold this PR up.

Filter Panel applies to all other panels in the same dashboard which use the same Entity datasource (they actually have to be using the React Entity Datasource). Nothing else special has to be done in any other panels.

Anywhere in another panel where you have a query that gets entity data, e.g. all nodes or alarms, it will also be filtered by any active Filter Panel selections.

For example in the case of the Alarm Table Panel, if you have a Filter Panel in the same dashboard, you can add an alarm filter in the Filter Panel, including a specific alarm attribute like 'severity', and any items selected in the Filter Panel will limit ('filter') which alarms are displayed in in the Alarm Table Panel.

Comment (synqotik) [reply docs/modules/panel_configuration/pages/index.adoc]: @emarsh1995 sounds good! Yes we can open another ticket if needed.

**Review by Bonrob2 — APPROVED** [review]
LGTM! Thank you!

**Review by mmahacek — APPROVED** [review]

**Pull request merged**
